packages/expo-cli/src/commands/build/ios/utils/image.js

```diff
@@ -0,0 +1,37 @@
+import fs from 'fs-extra';
+import { PNG } from 'pngjs';
+import { XDLError, ErrorCode } from 'xdl';
+
+export async function ensurePNGIsNotTransparent(imagePath) {
+  let hasAlreadyResolved = false;
+  return new Promise((res, rej) => {
+    fs.createReadStream(imagePath)
+      .pipe(new PNG({ filterType: 4 }))
+      .on('metadata', ({ colorType }) => {
+        if (colorType !== 6) {
+          hasAlreadyResolved = true;
```

Comment:
Should the read stream be destroyed at this point to prevent unnecessarily reading rest of the file?